pkg/kubeconfig/kubeconfig.go

```diff
@@ -37,3 +41,40 @@ func NewMissingFileError(path string) error {
 func (m missingFileError) Error() string {
 	return fmt.Sprintf("kubeconfig file missing: path=%v", m.Path)
 }
+
+// Validate checks that kubeconfig data is readable.
+//
+// This does not validate the values contained within; just that the data has
+// the basic structure of a kubeconfig.
+func Validate(r io.Reader) error {
+	kubeConfigData, err := ioutil.ReadAll(r)
+	if err != nil {
+		return err
+	}
+
+	// I wish I knew why the k8s authors avoid Reader/Writer interfaces so
+	// much. :(
+	if err := validator(kubeConfigData); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// validator can be overridden for easier testing.
+var validator = func(b []byte) error { _, err := clientcmd.Load(b); return err }
```

Comment:
nit: maybe overkill but maybe pass the validation as an option to the validate method? just trying to avoid global state

Reply:
Lemme look at this and see what makes sense. I def. don't love the package-level scope either.